### PR TITLE
:sparkles: Added truth table helpers for 3-input functions

### PR DESCRIPTION
**Pull request by marcelwa**
## Description

This PR adds several helper functions to create truth tables for some 3-input Boolean functions. The selection is based on D. S. Marakkalage et al. [*"Three-Input Gates for Logic Synthesis"*](https://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=9233431) in TCAD, 2021.

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [x] I have added appropriate tests and documentation.
- [ ] I have added a changelog entry.
- [x] I have created/adjusted the Python bindings for any new or updated functionality.
- [ ] I have made sure that all CI jobs on GitHub pass.
- [x] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

File: include/fiction/utils/truth_table_utils.hpp
Comment (Drewniok) [inline include/fiction/utils/truth_table_utils.hpp:1]: We should update ``docs/utils/utils.rst``.

**Review by Drewniok — CHANGES_REQUESTED** [review]

File: docs/utils/utils.rst [hunk 1]
```diff
@@ -52,7 +52,20 @@ Truth Table Utils
...
+        .. doxygenfunction:: fiction::create_and3_tt
+        .. doxygenfunction:: fiction::create_xor_and_tt
+        .. doxygenfunction:: fiction::create_or_and_tt
+        .. doxygenfunction:: fiction::create_onehot_tt
         .. doxygenfunction:: fiction::create_maj_tt
+        .. doxygenfunction:: fiction::create_gamble_tt
+        .. doxygenfunction:: fiction::create_dot_tt
+        .. doxygenfunction:: fiction::create_ite_tt
+        .. doxygenfunction:: fiction::create_and_xor_tt
+        .. doxygenfunction:: fiction::create_and_xor3_tt
```
Comment (Drewniok) [inline docs/utils/utils.rst:68]: I think this is a copy/paste mistake
```suggestion
        .. doxygenfunction:: fiction::create_xor3_tt
```

Comment (marcelwa) [reply docs/utils/utils.rst]: Good eyes! Many thanks :pray:

File: docs/utils/utils.rst
[the remark this answers is not in the document]
Comment (marcelwa) [reply docs/utils/utils.rst:1]: Of course. Sorry for the oversight. It's on me.

**Review by Drewniok — APPROVED** [review]
LGTM! Thank you! 🙏

**Pull request merged**